doc/source/guide/chambolle_pock_guide.rst

```diff
@@ -0,0 +1,176 @@
+Problem setup
+^^^^^^^^^^^^^
+The first step in the problem setup is the definition of the spaces in which we want to solve the problem.
+In this case, we use an :math:`L^2` space on the square :math:`[0, 100] \times [0, 100]`.
+We choose 256 discretization points per axis:
+
+.. code-block:: python
+
+    >>> space = odl.uniform_discr(min_pt=[0, 0], max_pt=[100, 100], shape=[256, 256])
+
+In real problems, the data :math:`y` would be given by some measurement, but for the purpose of testing the solver, we generate data by creating a modified `Shepp-Logan phantom <https://en.wikipedia.org/wiki/Shepp%E2%80%93Logan_phantom>`_ and adding 10% Gaussian noise:
+
+.. code-block:: python
+
+    >>> phantom = odl.phantom.shepp_logan(space, modified=True)
+    >>> data = phantom + odl.phantom.white_noise(space) * 0.1
+
+We now need to define the forward operator :math:`L`, which we do one constituent at a time:
+
+.. code-block:: python
+
+    >>> ident = odl.IdentityOperator(space)
+    >>> grad = odl.Gradient(space)
+
+To create :math:`L`, we use the `BroadcastOperator` class as mentioned above:
+
+.. code-block:: python
+
+    >>> L = odl.BroadcastOperator(ident, grad)
+
+We can now proceed to the problem specification.
+This step requires us to specify the functionals :math:`f` and :math:`g`, where the former is the `SeparableSum` of the squared :math:`L^2` distance to :math:`y` and the (vectorial) :math:`L^1` norm.
+These functionals are available in ODL as `L2NormSquared` and `L1Norm`, respectively:
+
+.. code-block:: python
+
+    >>> l2_norm_squared = odl.solvers.L2NormSquared(space).translated(data)
+    >>> l1_norm = 0.0003 * odl.solvers.L1Norm(grad.range)
+    >>> f = odl.solvers.SeparableSum(l2_norm_squared, l1_norm)
+
+.. note::
+    We don't need to take extra care of the :math:`L^1` norm being a vectorial norm since `L1Norm` also works on product spaces.
+
+Finally, we define the functional for the nonnegativity constraint, available as the functional `IndicatorNonnegativity`:
+
+    >>> g = odl.solvers.functional.IndicatorNonnegativity(space)
+
+Calling the solver
+^^^^^^^^^^^^^^^^^^
+Now that the problem is set up, we need to select some optimization parameters.
+For the Chambolle-Pock method, there is one main rule that we can use:
+The product of the primal step :math:`\tau`, the dual step :math:`\sigma` and the squared operator norm :math:`\|L\|^2` has to be smaller than 1, :math:`\tau \sigma \|L\|^2 < 1`.
+Apart from this, there are no clear rules on how to select :math:`\tau` and :math:`\sigma` -- basically we're left with trial and error.
+We decide to pick them both equal to :math:`1 / \|L\|^2`.
+To calculate an estimate of the operator norm, we have the tool `power_method_opnorm` which performs the simple `power iteration <https://en.wikipedia.org/wiki/Power_iteration>`_ to approximate the largest singular value of :math:`L`:
+
+.. code-block:: python
+
+   >>> op_norm = 1.1 * odl.power_method_opnorm(L, maxiter=4, xstart=phantom)
```

Comment:
But that happens to work only in this case since the operator domain and range are the same. I find it misleading to use a range element as input to the norm estimator.
Or - if we want to do that - we could accept a range element and start with the adjiont evaluation. Would be a new issue, though.